src/deep-shape-grammars/deep-shape-grammars.py

```diff
@@ -115,6 +117,65 @@ def new_grammar( self ):
 
             print( sel.bounds( image ) )
 
+            def get_image():
+                im = Gimp.get_pdb().run_procedure( 'gimp-get-images', [ ] )
+                image = im.index( 2 ).data[ 0 ]
+
+                return image
+
+            def save_image(img, layer, path):
+                return Gimp.get_pdb().run_procedure( 'file-png-save', [
+                    GObject.Value( Gimp.RunMode, Gimp.RunMode.NONINTERACTIVE ),
+                    GObject.Value( Gimp.Image, img ),
+                    GObject.Value( GObject.TYPE_INT, 1 ),
+                    GObject.Value( Gimp.ObjectArray, Gimp.ObjectArray.new( Gimp.Drawable, [ layer ], False ) ),
+                    GObject.Value( Gio.File, Gio.File.new_for_path( path ) ),
+                    GObject.Value( GObject.TYPE_BOOLEAN, 0 ),
+                    GObject.Value( GObject.TYPE_INT, 2 ),
+                    GObject.Value( GObject.TYPE_BOOLEAN, True ),
+                    GObject.Value( GObject.TYPE_BOOLEAN, False ),
+                    GObject.Value( GObject.TYPE_BOOLEAN, True ),
+                    GObject.Value( GObject.TYPE_BOOLEAN, True ),
+                    GObject.Value( GObject.TYPE_BOOLEAN, True ),
+                ])
+
+            def get_image_selection_bounds(img):
+                bounds = Gimp.get_pdb().run_procedure( 'gimp-selection-bounds', [
+                    GObject.Value( Gimp.Image, img )
+                ])
+                
+                return bounds.index(1), bounds.index(2), bounds.index(3), bounds.index(4), bounds.index(5)
+
+            def corp_image(img, x1, y1, x2, y2):
+                Gimp.get_pdb().run_procedure( 'gimp-image-crop', [
+                    GObject.Value( Gimp.Image, img ),
+                    x2 - x1,
+                    y2 - y1,
+                    x1,
+                    y1
+                ])
+
+            path = os.path.dirname( sys.argv[0] )
+            active_image = get_image()
+
+            # save org image
+            org_img = active_image.duplicate()
+            org_layer = org_img.merge_visible_layers( Gimp.MergeType.CLIP_TO_IMAGE )
+            org_img_path = f'{path}/image.png'
+            save_image(org_img, org_layer, org_img_path )
+
+            # save selection image
+            selection_img = active_image.duplicate()
+            selection_exists, x1, y1, x2, y2 = get_image_selection_bounds(selection_img)
```

Comment:
`selection_exists` indicates whether selection is present or not. we should probably alert user if `selection_exists  = false` as that means they did not select any area